packages/access-client/src/agent-use-cases.js

```diff
@@ -188,8 +191,12 @@ export async function waitForDelegationOnSocket(access, opts) {
 }
 
 /**
- * @typedef {{signal?: AbortSignal }} AuthorizationWaiterOpts
- * @typedef {(accessAgent: AccessAgent, opts: AuthorizationWaiterOpts) => Promise<Iterable<Ucanto.Delegation>> } AuthorizationWaiter
+ * @template {unknown} [T={}]
+ * @typedef {{ signal?: AbortSignal } & T} AuthorizationWaiterOpts
+ */
+/**
+ * @template {unknown} [U={}]
```

Comment:
```suggestion
 * @template [U={}]
```

Unless you explicitly need to constrain the type, it's usually best to omit it

Reply:
oh yea agree - for some reason I thought it was giving me errors without this, but it's working now, whatever I was seeing must have been a red herring, good catch!